src/lnp/application/payment/types.rs

```diff
@@ -425,7 +427,7 @@ impl StrictDecode for ShortChannelId {
     }
 }
 
-impl Strategy for ShortChannelId {
+impl LNPEncodingStrategy for ShortChannelId {
```

Comment:
This is a bit against rust language API guidelines: crate paths _are_ valid names, and we need to avoid uppercase sequences in type names (i.e. `LNP`). So `lnp::encoding::Strategy` still will be much better — and no need for an `as` import